src/modules/flow/http-server/http-server.c

```diff
@@ -713,6 +714,167 @@ float_process_cb(struct http_data *mdata, const struct sol_flow_packet *packet)
     return sol_flow_packet_get_drange(packet, &mdata->value.d);
 }
 
+static void
+rgb_send_packet_cb(struct http_data *mdata, struct sol_flow_node *node)
+{
+    sol_flow_send_rgb_packet(node,
+        SOL_FLOW_NODE_TYPE_HTTP_SERVER_RGB__OUT__OUT, &mdata->value.rgb);
+}
+
+static int
+rgb_response_cb(struct http_data *mdata, struct sol_buffer *content, bool json)
+{
```

Comment:
shouldn't you check the `json`? what you provide looks like the json return. I've noticed that other types, when complex, return the main value, something that doesn't exist for RGB... so I'm bit unsure on how to proceed, eventually we may force this return and change the Content-type? In this case we'd need to return something to notify a different type is to be used.

Another option is to return a text-like representation of color, like %g,%g,%g, that is, normalized 0-1 value of color components (red / red_max...).

It's basically your call, but document the behavior in the node description.